src/App.tsx

```diff
@@ -14,11 +15,97 @@ const goodsFromServer: string[] = [
   'Garlic',
 ];
 
-const App: React.FC = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+export class App extends React.Component {
+  state = {
+    selectedGoods: [],
+  };
 
-export default App;
+  render() {
+    const { selectedGoods } = this.state;
+
+    const message = () => {
+      if (selectedGoods.length === 1) {
+        return `${selectedGoods.join('')} is selected`;
+      }
+
+      if (selectedGoods.length > 1) {
+        return `${selectedGoods.join(',')} are selected`;
+      }
+
+      return 'No goods selected';
+    };
+
+    return (
+      <>
+        <div className="App">
+          <h1 className="title">
+            Selected good:
+            <span className="red">
+              {message()}
+            </span>
+            <button
+              type="button"
+              className="button button__h1"
+              onClick={() => {
+                this.setState({ selectedGoods: [] });
+              }}
+            >
+              X
+            </button>
+          </h1>
+          <ul className="goods">
+            {goodsFromServer.map(good => (
+              <div key={good} className="good__block">
+                <li className={classNames(
+                  'good',
+                  {
+                    good__active: selectedGoods.includes(good as never),
```

Comment:
```suggestion
                    good__active: selectedGoods.includes(good),
```
bad practice to deceive TS